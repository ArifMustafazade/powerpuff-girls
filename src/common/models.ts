/** @interfaces */

export interface IApiHandlerResponse<T> {
    status: boolean;
    errorType?: string;
    errorMessage?: string;
    data?: T;
}

interface ILink {
    href: string;
}

interface INetwork {
    id: number;
    name: string;
    country: {
        name: string;
        code: string;
        timezone: string;
    };
}

export interface IEpisode {
    id: number;
    url: string;
    name: string;
    season: number;
    number: number;
    type: string;
    airdate: string;
    airtime: string;
    airstamp: string;
    runtime: number;
    image: {
        medium: string;
        original: string;
    };
    summary: string;
    _links: {
        self: ILink;
    };
}

export interface IShowDetails {
    id: number;
    url: string;
    name: string;
    type: string;
    language: string;
    genres: string[];
    status: string;
    runtime: number;
    premiered: string;
    officialSite: string;
    schedule: {
        time: string;
        days: string[];
    };
    rating: {
        average: number;
    };
    weight: number;
    network: INetwork;
    webChannel: string;
    externals: {
        tvrage: number;
        thetvdb: number;
        imdb: string;
    };
    image: {
        medium: string;
        original: string;
    };
    summary: string;
    updated: number;
    _links: {
        self: ILink;
        previousepisode: ILink;
    };
}
