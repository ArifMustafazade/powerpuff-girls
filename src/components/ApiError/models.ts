/** @interfaces */

export interface IApiErrorProps {
    errorMessage: string;
}
