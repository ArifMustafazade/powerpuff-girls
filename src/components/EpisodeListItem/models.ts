// interface
import { IEpisode } from '../../common/models';

/** @interfaces */

export interface IEpisodeListItemProps {
    episode: IEpisode;
}
