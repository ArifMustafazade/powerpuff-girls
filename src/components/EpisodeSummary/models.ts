// interface
import { IEpisode } from '../../common/models';

/** @interfaces */

export interface IEpisodeSummaryProps {
    episode?: IEpisode;
}
