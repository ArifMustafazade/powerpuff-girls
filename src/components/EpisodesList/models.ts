// interfaces
import { IEpisode } from '../../common/models';

/** @interfaces */
export interface IEpisodesListProps {
    episodes: IEpisode[];
    loading: boolean;
    error: boolean;
    errorMessage: string;
}
