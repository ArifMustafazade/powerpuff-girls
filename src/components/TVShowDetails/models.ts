// interfaces
import { IShowDetails } from '../../common/models';

/** @interfaces */
export interface ITVShowDetailsProps {
    details: IShowDetails;
    error: boolean;
    errorMessage: string;
    loading: boolean;
}
