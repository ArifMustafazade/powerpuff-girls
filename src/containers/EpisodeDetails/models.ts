// interfaces
import { RouteComponentProps } from 'react-router-dom';
import { IEpisodesState } from '../../store/models';
import { IEpisode } from '../../common/models';

/** @interfaces */
export interface IEpisodeDetailsProps extends RouteComponentProps<{ id: string }> {
    episodesList: IEpisodesState;
}

export interface IEpisodeDetailsState {
    episode: IEpisode;
    error: boolean;
    errorMessage: string;
    loading: boolean;
}
