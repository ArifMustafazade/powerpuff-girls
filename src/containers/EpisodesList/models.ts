// interfaces
import { IEpisodesState } from '../../store/models';

/** @interfaces */
export interface IEpisodesListProps {
    episodesList: IEpisodesState;
    onGetEpisodesList: () => void;
}
