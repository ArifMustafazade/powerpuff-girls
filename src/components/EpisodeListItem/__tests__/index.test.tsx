import React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';

// interface
import { IEpisode } from '../../../common/models';
import { IEpisodeListItemProps } from '../models';

// components
import { EpisodeListItem } from '../';

/** tests */
describe('<EpisodeListItem />', () => {
    let wrapper: ShallowWrapper<IEpisodeListItemProps>;

    const mockEpisode: IEpisode = {
        id: 1,
        url: 'http://www.tvmaze.com/episodes/1/under-the-dome-1x01-pilot',
        name: 'Pilot',
        season: 1,
        number: 1,
        type: 'regular',
        airdate: '2013-06-24',
        airtime: '22:00',
        airstamp: '2013-06-25T02:00:00+00:00',
        runtime: 60,
        image: {
            medium: 'http://static.tvmaze.com/uploads/images/medium_landscape/1/4388.jpg',
            original: 'http://static.tvmaze.com/uploads/images/original_untouched/1/4388.jpg',
        },
        summary:
            "<p>When the residents of Chester's Mill find themselves trapped under a massive transparent dome with no way out, they struggle to survive as resources rapidly dwindle and panic quickly escalates.</p>",
        _links: {
            self: {
                href: 'http://api.tvmaze.com/episodes/1',
            },
        },
    };
    beforeEach(() => {
        wrapper = shallow(<EpisodeListItem episode={mockEpisode} />);
    });

    test('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should render with correct season', () => {
        wrapper.setProps({
            episode: { ...mockEpisode, season: 3 },
        });

        expect(wrapper.find('.episodes__episode-season').text()).toBe('Season: 3');
    });

    test('should render with correct number', () => {
        wrapper.setProps({
            episode: { ...mockEpisode, number: 7 },
        });

        expect(wrapper.find('.episodes__episode-number').text()).toBe('Number: 7');
    });
});
