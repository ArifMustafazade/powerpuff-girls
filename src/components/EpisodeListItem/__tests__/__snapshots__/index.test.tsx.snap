// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EpisodeListItem /> should match snapshot 1`] = `
<Link
  className="episodes__episode"
  to="/episodes/1"
>
  <p
    className="episodes__episode-season"
  >
    Season: 1
  </p>
  <p
    className="episodes__episode-number"
  >
    Number: 1
  </p>
</Link>
`;
